### ISSUE TITLE
REVEAL: Remove `bio_data` assumption from content types

### ISSUE THREAD

**Issue by laceysanderson** [tripal/tripal#1565]
# Core Tripal 4 Development Task

## Current Issue Description

We have used bio_data in the URLs and a couple of other places in our Tripal Content. However, this is not really necessary and not ideal for researcher friendliness 🙈 

In #1413 Stephen started to remove this assumption, but it was not fully completed and may have been the change that broke the routing in that PR (we were never able to confirm this though).

Specific places to remove Bio_data from:
- content type machine name. 
    - It would be easier if these were named in a human readable way (e.g. organism instead of bio_data_1) to make it easier for extension modules to alter specific content types. You should not rely on the specific bio_data_### as this is dependant upon order of creation. As such you currently should look up the content type using other means or look up the bio_data_# mapping first. This can be error prone...
    - tripal_chado/config/install/tripal.tripalfield_collection.default_chado.yml and tripal_chado/config/install/tripal.tripalentitytype_collection.default_chado.yml
    - TripalFieldCollection.php and TripalEntityTypeCollection.php
- Routing / URLs
    - the URL within the structure: edit, manage field, manage display, etc.
 
## Background

The background for this issue is a cautionary tale: Once upon a time a PR was made, #1413, with so many great changes for Tripal 4. This PR was focused on moving content type + field creation into YAML files for the good of all. This was a large task in and of itself but as such tasks have a habit of doing... it picked up many additional bug fixes and features as it grew. The developer was testing locally both manually and running the automated tests they developed just as they should but forgot to push out to github as often as recommended 😨 This caused the developer to miss a test failure in unrelated code within their beautiful PR 😱 As the PR had become an inter-related collection of fixes by the point it came up for review and the unrelated failing test was found... it became a nightmare to debug and though the developer and their friends tried, they could not find the bug 😭 🗿  This meant that the PR had to be redone with careful testing and often github pushing to ensure the bug did not resurface... This processed birthed PR #1475 and this collection of issues.

tl;dr;
- PR #1413 included a large number of fixes but needed to be closed due to a test failure in unrelated code which could not be fixed
- PR #1475 brought back the core functionality the PR originally meant to cover (Content Types + Fields created via YAML) but could not bring back all of the fixes from the original PR.
- This issue has been created to mark the place of one of these fixes to be brought back.

## Process:
- Look at PR #1413 and try to find the code that made this particular fix. 
- Create a new branch from 4.x using the branch name above
- Copy small bits of the changes from PR #1413 into this branch and commit. Only move pieces related to the fix!
  - When committing you will want to give both yourself and Stephen credit for the code! See following for how to do that (Co-authored commit).
- Push to the branch after each addition to make sure there are no test failures 😬 

### Co-Authored Commits

See [these github docs for more detail](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors).

For each commit including any of Stephens original code use the following. Note that you do not close the quotation after your typical commit message but rather enter two empty lines and add the co-author information.

```bash
git commit -m "Your typical commit message


Co-authored-by: Stephen Ficklin <spficklin@gmail.com>
Co-authored-by: YOUR-NAME <YOUR-NAME@EXAMPLE.COM>"
```

Example: https://github.com/tripal/tripal/pull/1475/commits/af6bc9b18b484fa7d24eda73bf2b73b6256d57cd

If needed you can alter your commit message to add this after the fact. This is easiest if done before pushing to github.

**Comment by spficklin** [comment]
I started working on this.  Branch is `7v4g1-1565-Remove_bio_data`

**Comment by spficklin** [comment]
Fixed with reently merged PR

**Comment by dsenalik** [comment]
Fixed some more with recently merged PR #1601
